### DISCUSSION TITLE
Suggested way of implementing multi zones with sessions

### DISCUSSION THREAD

**Discussion by asaldivar** [vercel/next.js#12907] [Help]
Hi,

I'm implementing [multi zones](https://github.com/zeit/next.js/tree/canary/examples/with-zones) with [firebase sessions](https://github.com/zeit/next.js/tree/canary/examples/with-firebase-authentication) and was wondering if anyone has a suggestion for how to handle sessions across zones? I'd like one zone to act as an entry point for authentication/session creation which also provides a view for links to other zones once User is authenticated. The other zones will serve as different services User can use. Curious if anyone has done this already or has a recommendation for how to store the sessions as well as how to work with sessions across zones locally as well as in prod. Any thoughts are appreciated.

Thanks!

**Comment by stopyransky** [comment]
I also have similar question relating to setting up next.js project with zones and Auth0.

My case is I have default 'website', having routes `/` and `/blog` and then restricted `/app` for only authenticated users and `/admin` which is only for authenticated users but also with admin role. 

So I have 3 zones: default, app and admin. For each there is a separate repository. 
These repos are bound with vercel.json in top directory (just like in [this example](https://github.com/vercel/next.js/tree/canary/examples/with-zones)).

Here is also my question - how do i share user session / credentials among the zones? 
I assume i will have to build some [custom server](https://nextjs.org/docs/advanced-features/custom-server) but this seems redundant to have per each zone... and I am not sure how to communicate user session data between zones ?

**Reply by GxDesign** [reply]
If your zones all run on the same domain, your session is automatically shared. If your site is whitelisted for a backend that handles the session, you can use credentials: 'include' for data fetching. The example was built for vercel deployment and vercel currently does not have true monorepo support for zones on different domains.
